### PR TITLE
🤖 Sentinel: [Killed mutant in WalRingBuffer backpressure logic]

### PR DESCRIPTION
**Pull request by madmax983**
**🧬 Mutants Found:** 1 surviving mutant in `src/storage/wal/ring_buffer.rs` (replaced `<=` with `<` in backpressure check).
**🎯 Tests Added/Strengthened:** Added `test_sentry_backpressure_hang_protection` to `mod sentry_tests`.
**Diagnosis:** The mutation `distance_behind <= self.capacity` -> `distance_behind < self.capacity` caused `try_append` to enter an infinite loop (hang) instead of returning `Err` when a previous writer stalled and the buffer was full (distance == capacity). Existing tests did not cover this specific "stalled writer" boundary condition.
**Kill Shot:** The new test manually simulates a stalled writer (advancing `write_pos` without updating sequence) and asserts that `try_append` returns `Err` (backpressure) instead of hanging. This test passes on the original code and fails (times out) on the mutant.
**📊 Kill Rate:** Improved coverage for `WalRingBuffer` concurrent edge cases.

---
*PR created automatically by Jules for task [10677579489204376379](https://jules.google.com/task/10677579489204376379) started by @madmax983*

### CODE REVIEW

File: src/storage/wal/ring_buffer.rs [hunk 1]
```diff
@@ -1457,4 +1457,54 @@ mod sentry_tests {
...
+        // current_seq = 0 (Writer A stalled).
+        // distance_behind = 2 - 0 = 2.
+        // capacity = 2.
+        // distance_behind == capacity.
+
+        // If the code is correct (<= capacity), it should detect "Buffer Full" and return Err after spinning.
+        // If the mutant is present (< capacity), it will think "Another producer claimed it" and continue/loop forever.
+
+        let entry_c = PendingEntry::new_async(LSN(2), vec![]);
+        let result = buf.try_append(entry_c);
```
[the remark this answers is not in the document]
Comment (madmax983) [reply src/storage/wal/ring_buffer.rs]: > **P2 Badge Add timeout guard around non-terminating append check**
> 
> The new sentry test calls `buf.try_append(entry_c)` directly, but the regression it is trying to catch (`distance_behind < capacity`) turns that call into an infinite loop; without a timeout boundary, the test can hang the entire `cargo test` run instead of producing a deterministic failure. Wrapping this call in a bounded-time worker (or equivalent timeout assertion) would keep CI failures actionable when this bug reappears.
> 
> Useful? React with 👍 / 👎.

@jules
